src/Agent/NewRelic/Agent/Core/Logging/ProcessIdEnricher.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using NewRelic.SystemInterfaces;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace NewRelic.Agent.Core
+{
+    class ProcessIdEnricher : ILogEventEnricher
+    {
+        public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
+        {
+            logEvent.AddPropertyIfAbsent(propertyFactory.CreateProperty(
+                "pid", new ProcessStatic().GetCurrentProcess().Id));
```

Comment:
Do we need to cache this process lookup?

Reply:
Shouldn't be necessary. The code in `ProcessStatic` is already doing that.

Reply:
It's not necessary, but it's an additional allocation every time this enricher executes that could easily be reduced.

Reply:
Refactored to get process id from a static field.